### PR TITLE
v2.0 flutter

### PR DESCRIPTION
**Pull request by ksyeo1010**
- updated flutter binding and demo

### CODE REVIEW

File: sdk/flutter/lib/picovoice_error.dart [hunk 1]
```diff
@@ -9,75 +9,97 @@
...
-      throw new PvStatusIoError(errorMessage);
-    case PvStatus.INVALID_ARGUMENT:
-      throw new PvStatusInvalidArgumentError(errorMessage);
-    case PvStatus.STOP_ITERATION:
-      throw new PvStatusStopIterationError(errorMessage);
-    case PvStatus.KEY_ERROR:
-      throw new PvStatusKeyError(errorMessage);
-    case PvStatus.INVALID_STATE:
-      throw new PvStatusInvalidStateError(errorMessage);
+mapToPicovoiceException(Exception ex, String? message) {
```
Comment (laves) [inline sdk/flutter/lib/picovoice_error.dart:64]: can we wrap the inner exception as well?

File: sdk/flutter/lib/picovoice.dart [hunk 2]
```diff
@@ -119,25 +154,24 @@ class Picovoice {
   /// [frame] A frame of audio samples. The number of samples per frame can be attained by calling
   /// `.frameLength`. The incoming audio needs to have a sample rate equal to `.sample_rate` and be 16-bit linearly-encoded.
   /// Picovoice operates on single-channel audio.
-  void process(List<int> frame) {
+  void process(List<int> frame) async {
```
Comment (laves) [inline sdk/flutter/lib/picovoice.dart:157]: can we add a check for null frame and frame length here?

File: sdk/flutter/lib/picovoice.dart [hunk 3]
```diff
@@ -10,42 +10,68 @@
...
-import 'package:rhino/rhino_error.dart' as rhinoErr;
+import 'package:porcupine_flutter/porcupine.dart';
+import 'package:porcupine_flutter/porcupine_error.dart';
+import 'package:rhino_flutter/rhino.dart';
+import 'package:rhino_flutter/rhino_error.dart';
+
+typedef WakeWordCallback = Function();
+typedef InferenceCallback = Function(PicovoiceInference inference);
+
+class PicovoiceInference {
```
Comment (laves) [inline sdk/flutter/lib/picovoice.dart:21]: Can't we just use RhinoInference here? Why do we need to create a new class that duplicates it?

**Review by laves — CHANGES_REQUESTED** [review]
Can you update the flutter-clock demo as well?

File: sdk/flutter/lib/picovoice.dart (hunk 2 above)
Comment (ksyeo1010) [reply sdk/flutter/lib/picovoice.dart:132]: only check for frame length since null can't be passed it

**Review by laves — CHANGES_REQUESTED** [review]
flutter-clock demo updates?

**Review by laves — APPROVED** [review]

**Pull request merged**